src/test/java/bio/terra/service/resourcemanagement/google/BucketResourceTest.java

```diff
@@ -591,4 +592,19 @@ private GoogleProjectResource buildProjectResource() throws Exception {
         Map.of("test-name", "bucket-resource-test"),
         CollectionType.DATASET);
   }
+
+  /**
+   * @param baseName the desired prefix for a GCS bucket name
+   * @return a globally-unique GCS bucket name based off of the provided baseName
+   * @throws IllegalArgumentException if the resulting bucket name is known not to adhere to GCS
+   *     naming requirements (https://cloud.google.com/storage/docs/buckets#naming)
+   */
+  private String createBucketName(String baseName) throws IllegalArgumentException {
+    String globallyUniqueName = Names.randomizeName(baseName.toLowerCase());
+    assertThat(
```

Comment:
An assert here is fine but I wouldn't add `throws` or document it; IMO the point of `throws` is to tell a caller about exceptions they might want to catch. In this case the method is only used in tests and if a problem occurs we want the test to fail, so the throw exception will never be caught.

Also `assertThat()` will throw `AssertionError` not `IllegalArgumentException`, so I don't see where `IllegalArgumentException` would get thrown.

Reply:
That's an oversight from an earlier attempt where I considered throwing. I'll clean it up.